src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -1061,7 +1084,15 @@ public synchronized void handle(Message<T> event) {
         }
       } else {
         checkNextTick();
-        handler.handle(event);
+        MessageImpl abc = (MessageImpl) event;
+        metrics.beginHandleMessage(metric, abc.getSocket() == null);
+        try {
+          handler.handle(event);
+          metrics.endHandleMessage(metric, null);
```

Comment:
Everything lies down in the implementation of the metrics SPI. The EventBusMetrics implementation with Codahale has been modified to care only about the handlers that match a certain address.

When an handler is registered, it is matched against a list of permitted address : https://github.com/vert-x3/vertx-metrics/blob/eventbus-metrics/src/main/java/io/vertx/ext/metrics/impl/EventBusMetricsImpl.java#L99

When nothing matches then it returns null.

Then when the handler metric is null then nothing is happens:

https://github.com/vert-x3/vertx-metrics/blob/eventbus-metrics/src/main/java/io/vertx/ext/metrics/impl/EventBusMetricsImpl.java#L123
https://github.com/vert-x3/vertx-metrics/blob/eventbus-metrics/src/main/java/io/vertx/ext/metrics/impl/EventBusMetricsImpl.java#L136

It is fine grained indeed and the metric SPI implementation should really care about it. A broken SPI implementation can certainly make vertx behave badly.